pkg/cloud/ovirt/clients/machineservice.go

```diff
@@ -423,3 +458,77 @@ func (is *InstanceService) handleAffinityGroups(vm *ovirtsdk.Vm, cID string, ags
 	}
 	return nil
 }
+
+// getHostsInCluster will return a HostSlice of the hosts in a given cluster ID.
+func (is *InstanceService) getHostsInCluster(clusterId string) (*ovirtsdk.HostSlice, error) {
+	clusterService := is.Connection.SystemService().ClustersService().ClusterService(clusterId)
+	clusterGet, err := clusterService.Get().Send()
+	if err != nil {
+		return nil, errors.Errorf("failed to get the cluster: %v", err)
+	}
+	clusterName := clusterGet.MustCluster().MustName()
+	hostsInCluster, err := is.Connection.SystemService().HostsService().List().Search(
+		fmt.Sprintf("cluster=%s", clusterName)).Send()
+	if err != nil {
+		return nil, errors.Errorf("failed to get the list of hosts in the cluster: %v", err)
+	}
+	return hostsInCluster.MustHosts(), nil
+}
+
+// handleAutoPinning updates the VM after creation to set the auto pinning policy configuration.
+func (is *InstanceService) handleAutoPinning(id string, autoPinningPolicy ovirtsdk.AutoPinningPolicy) error {
+	vmService := is.Connection.SystemService().VmsService().VmService(id)
+	optimizeCpuSettings := !(autoPinningPolicy == ovirtsdk.AUTOPINNINGPOLICY_EXISTING)
```

Comment:
(autoPinningPolicy != ovirtsdk.AUTOPINNINGPOLICY_EXISTING)

So optimizeCpuSettings should be for disabled and adjust?

Reply:
No, we did missed the disabled option by using it. When we call the service(API) in this case it is only for existing/adjust. When you have disabled, just don't do anything. I did outside the function now.

Reply:
So I don't understand why are you using:
optimizeCpuSettings := !(autoPinningPolicy == ovirtsdk.AUTOPINNINGPOLICY_EXISTING)

that means that if autoPinningPolicy is adjust then optimizeCpuSettings will be false

Reply:
optimizeCpuSettings := autoPinningPolicy == ovirtsdk.AUTOPINNINGPOLICY_ADJUST
will solve it.
If we are in ADJUST then -> true
if we are having something else(only existing is possible) -> false